src/gsy_e/models/state.py

```diff
@@ -894,19 +896,109 @@ def get_results_dict(self, current_time_slot: DateTime) -> dict:
 class HeatPumpState(StateInterface):
     """State for the heat pump strategy."""
 
+    def __init__(self, initial_temp_C: float, slot_length: duration):
+        # the defaultdict was only selected for the initial slot
+        self._storage_temp_C: Dict[DateTime, float] = defaultdict(lambda: initial_temp_C)
+        self._min_energy_demand_kWh: Dict[DateTime, float] = {}
+        self._max_energy_demand_kWh: Dict[DateTime, float] = {}
+        # buffers for increase and  decrease of storage
+        self._temp_decrease_K: Dict[DateTime, float] = defaultdict(lambda: 0)
```

Comment:
Could you please describe why don't we operate on the `_storage_temp_C` directly and separate temp decrease/increase here? IMO, It would be much simpler to handle just one dictionary in `HeatPumpEnergyParameters` instead of three of them. This could also lead to removal of many setter/getter methods in `HeatPumpState` and simpler serializing/deserializing of the state.

Reply:
Your first reason convinced me. Thanks a lot